source/opt/loop_descriptor.h

```diff
@@ -177,6 +221,16 @@ class Loop {
   // computed only when needed on demand.
   BasicBlockListTy loop_basic_blocks_;
 
+  BasicBlockOrderedListTy loop_basic_blocks_in_order_;
+
+  ir::IRContext* ir_context_;
+  opt::DominatorAnalysis* dom_analysis_;
```

Comment:
not sure you need those, ask them as parameter if really needed (which I think you won't)